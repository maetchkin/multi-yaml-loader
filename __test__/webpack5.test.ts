import type { Compiler, Stats }           from 'webpack5';
import getWebpack5Compiler                from './getWebpack5Compiler';
import sanitizeSnapshot                   from './sanitizeSnapshot';

const compile =  (compiler: Compiler) =>
    new Promise(
        (resolve, reject) => {
            compiler.run(
                (error, stats) => {
                    error ? reject(error) : resolve(stats)
                }
            );
        }
    )
;

test(
    'webpack5',
    () => {
        const compiler = getWebpack5Compiler('documents/simple.yaml', {keepFiles:true, space:true});
        return compile(compiler)
            .then(
                (stats: Stats) => {
                    //console.log('stats', stats.toString() );
                    const code = stats?.compilation?.assets;
                    expect(sanitizeSnapshot(code)).toMatchSnapshot("documents-simple");
                }
            )
        ;
    }
);

afterAll(
    async () => {
        await new Promise(
            resolve =>
                setTimeout(
                    () => resolve(void(0)),
                    2500
                )
        ); // avoid jest open handle error
    }
);
