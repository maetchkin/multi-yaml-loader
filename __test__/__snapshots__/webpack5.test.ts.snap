// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack5: documents-simple 1`] = `
{
  "main.bundle.js": {
    "_size": 3450,
  },
}
`;
